### PR TITLE
Canvas: Match connection anchor points to elements

### LINKED ISSUE #84845: Canvas: Make connection point match the shape
Right now connection points located on shape wrapper, like this:
Screenshot 2024-03-20 at 17 23 41

but it would be better to match connection point with actual shape, e.g. like this:
Screenshot 2024-03-20 at 17 24 27

### PR DESCRIPTION
**Pull request by nmarrs**
TODO

- [x] Fix cloud anchor points to actually match shape
- [x] Fix parallelogram phantom point showing up
- [x] Better align / space top / bottom parallelogram points
- [x]  Figure out how to make anchor points show up correctly on initial hover vs needing to first select and then unselect element
- [x] Test and ensure no regressions

Ellipse

Screenshot 2024-03-30 at 2 12 21 AM

Triangle
Screenshot 2024-03-30 at 2 12 45 AM

Parallelogram
Screenshot 2024-04-22 at 7 36 59 PM

Cloud
Screenshot 2024-04-22 at 7 37 29 PM

Fixes https://github.com/grafana/grafana/issues/84845

### CODE REVIEW

**Review by nmarrs — COMMENTED** [review]
I think this is good to go :) thank you @Develer for pushing this over the finish line!!

**Review by baldm0mma — APPROVED** [review]
Love the comments! LGTM!

File: public/app/features/canvas/elements/cloud.tsx [hunk 1]
```diff
@@ -184,6 +184,24 @@ export const cloudItem: CanvasElementItem = {
         },
       });
   },
+
+  customConnectionAnchors: [
+    { x: -0.58, y: 0.63 }, // Top Left
```
Comment (codeincarnate) [inline public/app/features/canvas/elements/cloud.tsx:189]: Might be better to have property names? Seems a little cleaner and could be a good follow-up

**Review by codeincarnate — APPROVED** [review]
Tested it out and looks good!

**Pull request merged**